tests/conftest.py

```diff
@@ -1173,16 +1184,16 @@ async def _async_init_recorder_component(hass, add_config=None, db_url=None):
 
 @pytest.fixture
 async def async_setup_recorder_instance(
-    recorder_db_url,
-    enable_nightly_purge,
-    enable_statistics,
-    enable_statistics_table_validation,
-) -> AsyncGenerator[SetupRecorderInstanceT, None]:
+    recorder_db_url: str,
+    enable_nightly_purge: bool,
+    enable_statistics: bool,
+    enable_statistics_table_validation: bool,
+) -> AsyncGenerator[RecorderInstanceGenerator, None]:
```

Comment:
`SetupRecorderInstanceT` (defined in `common.py`) is no longer used in `conftest.py`, but it is still used in some of the tests.
I plan to remove it fully in a follow-up PR, but I want to avoid a conflict in case some recorder tests still need to be shipped in a patch release.

cc @emontnemery / @bdraco